### ISSUE TITLE
FSE: The navigation block does not allow additional CSS classes

### ISSUE THREAD

**Issue by apeatling** [Automattic/wp-calypso#34563]
1. Add the navigation block to the header
2. Open the block sidebar
3. Add a class in the "Additional CSS Classes" section.

The block will stop rendering and show an error:

<img width="1072" alt="Screen Shot 2019-07-09 at 12 56 45 PM" src="https://user-images.githubusercontent.com/1464705/60918858-2646d780-a249-11e9-9ba7-f43aa2c0b3a5.png">

**Comment by noahtallen** [comment]
Investigating this a bit, and I see this error when trying to set the class on only the nav block:
> GET localhost:8888/index.php?rest_route=%2Fwp%2Fv2%2Fblock-renderer%2Fa8c%2Fnavigation-menu&context=edit&attributes%5BthemeLocation%5D=menu-1&attributes%5BclassName%5D=**WHATEVER CLASSNAME YOU SET**&post_id=30&_locale=user 400 (Bad Request)

**Comment by noahtallen** [comment]
I discovered that the error does not appear if we define 'className' as one of the possible attribute types during register block type. My guess is that the API is doing a type check on the attributes and throwing an error because `className` doesn't exist in the types we specified as "allowed".

If we change from:
https://github.com/Automattic/wp-calypso/blob/6572a4bb497cac9f6ae14a53f6dc4a362d1b6bf9/apps/full-site-editing/full-site-editing-plugin/full-site-editing/class-full-site-editing.php#L323-L334

To something more like:
```php
register_block_type(
	'a8c/navigation-menu',
	array(
		'attributes'      => [
			'themeLocation' => [
				'default' => 'main-1',
				'type'    => 'string',
			],
			'className' => [
				'default' => '',
				'type'    => 'string'
			]
		],
		'render_callback' => 'render_navigation_menu_block',
	)
);
```
Then the bug is fixed and the class name saves correctly. I also looked into making a similar change to `blocks/navigation-menu/index.js:registerBlockType`, but the bug still happens if you do it there.

I'm also not sure how to specify that the className is actually optional. That solution still seems to handle that case, though.
